TestBenches/TrackMerger_test.cpp

```diff
@@ -0,0 +1,61 @@
+// TrackMerger test bench
+#include "TrackMergerTop.h"
+#include "FileReadUtility.h"
+#include "hls_math.h"
+
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <algorithm>
+#include <iterator>
+
+const int nevents = 100;
+
+using namespace std;
+
+int main(){
+    //Error counter
+    int err_count = 0;
+    
+  // Input memories
+  TrackFitMemory inputTracks;
+
+  // Output memories
+  TrackFitMemory outputTracks;
+
+  // Open input files
+  ifstream fin_inputTracks("../../../../../emData/TM/TM_L1L2/TrackFit_BT_L1L2_04.dat");
+  assert(fin_inputTracks.good());
+  //ifstream fout_outputTracks("../../../../../emData/TM/TM_L1L2/TrackFit_BT_L1L2_04.dat");
+  ifstream fout_outputTracks("../../../../../emData/TM/TM_L1L2/TrackFit_PT_L1L2_04.dat");
+  assert(fout_outputTracks.good());
+
+  // Loop over events
+  for (int ievt = 0; ievt < nevents; ++ievt) {
+    cout << "Event: " << dec << ievt << endl;
+
+    outputTracks.clear();
+
+    // Read in next event from input
+    writeArrayFromFile<TrackFiMemory> (inputTracks, fin_inputTracks, ievt);
+
+    // Set bunch crossing
+    BXType bx = ievt;
+
+    // Unit under test
+    TrackMergerTop(bx, inputTracks, outputTracks);
```

Comment:
The way you are calling `TrackMergerTop` doesn't match how it's defined in TrackMergerTop.h/cc. You need to convert `inputTracks` into the arrays of input track/stub words that the top function expects and define arrays for the output words. Then you can fill `outputTracks` from the arrays of output words just like the test-bench for the TrackBuilder does here:
https://github.com/cms-L1TK/firmware-hls/blob/3ebd273bdff13cf027166db804c7ca1fc38791a1/TestBenches/TrackBuilder_test.cpp#L252-L266

Reply:
This looks good now, except you need to pass just the variable names, since you're calling `TrackMergerTop` and not defining it, e.g. `const BXType bx` → `bx`. Same thing applies when you call `TrackMerger` in TrackMergerTop.cc.

You should also make sure your definitions of `TrackMergerTop` and `TrackMerger` are consistent between all your header and source files.